### PR TITLE
fix(Geocoder): prevent conditional hook calls in Geocoder

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Add `StandaloneGeocoder` and `ControlledGeocoder` components to prevent conditional React hook calls

#### Issue #, if available
NA

#### Description of how you validated changes
- ran integ tests
#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ErikCH — APPROVED** [review]
👍

File: packages/react/src/components/Geo/Geocoder/index.tsx [hunk 1]
```diff
@@ -45,29 +45,37 @@ export const Geocoder = ({
...
+
+  return null;
+};
+
+const StandaloneGeocoder = (props: GeocoderProps) => {
+  const hasMounted = useRef(false);
+
+  useEffect(() => {
+    if (!hasMounted.current) {
+      (createAmplifyGeocoder(props) as unknown as GeocoderControl).addTo(
```
Comment (ErikCH) [inline packages/react/src/components/Geo/Geocoder/index.tsx:70]: I've never seen `as unknown as GeocoderControl` . That's interesting way of doing it.

**Review by joebuono — APPROVED** [review]
LGTM

File: packages/react/src/components/Geo/Geocoder/index.tsx (hunk 1 above)
Comment (calebpollman) [reply packages/react/src/components/Geo/Geocoder/index.tsx:70]: I'm not quite sure why the casting was done this way and due to my lack of familiarity with the feature I chose to do the least impactful change for this PR. In my experience casting to unknown is only required when casting from type A to type B where type A has nothing in common with type B

**Pull request merged**